keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,649 @@
+non-goals of this KEP.
+
+### Goals
+
+* Introduce the ability for DRA to advertise extended resources listed in a
+  ResourceSlice, and for the scheduler to consider them for allocation.
+
+* Enable application operators to use the existing extended resource limits to
+request for DRA resources.
+
+### Non-Goals
+
+* Extended resource support is not added just for easing the transition to DRA
+  for the short term. Its ease of use is one big advantage to keep it remaining
+  useful for the long term.
+
+## Proposal
+
+The basic idea is the following:
+
+1. Introduce a field `ExtendedResourceName` to `ResourceSlice` and `Device` to
+   allow cluster administrators to configure DRA device driver to advertise
+   certain devices as extended resource.
+1. Introduce a special `ResourceClaim` to keep track of device allocations for
+   extended resource.
+
+With these additions in place, the scheduler has everything it needs to support
+the dynamic allocation of devices to requests made through extended resource.
+kubelet and DRA driver has everything it needs to admit and pass the allocated
+devices to the pod to run.
+
+## Design Details
+
+### Resource Slice API
+The exact set of proposed API changes on Resource Slice can be seen below:
+```go
+// ResourceSliceSpec contains the information published by the driver in one ResourceSlice.
+type ResourceSliceSpec struct {
+	...
+
+	// The extended resource name for all the devices in the ResourceSlice
+	// advertised as
+	//
+	// +optional
+	ExtendedResourceName string
+}
+
+// Device represents one individual hardware instance that can be selected based
+// on its attributes. Besides the name, exactly one field must be set.
+// +k8s:deepcopy-gen=true
+type Device struct {
+	// Name is unique identifier among all devices managed by
+	// the driver in the pool. It must be a DNS label.
+	//
+	// +required
+	Name string `json:"name"`
+	...
+
+	// ExtendedResourceName is the extended resource name
+	// the device is advertised as. It must be a DNS label.
+	//
+	// +optional
+	ExtendedResourceName string
+}
+```
+
+The devices can be advertised with an extended resource name. The extended
+resource name can be specified on each individual device. Different
+devices can be advertised as different extended resource name, or not
+advertised as extended resource at all.
+
+Alternatively, the extended resource name can be specified at the
+`ResourceSlice` level, then all the devices in the resource slice are
+advertised as the given extended resource name. If a device has a different
+extended resource name than that given in the `ResoureSlice`, the device's
+extended resource name is used for that device.
+
+### Resource Claim API
+There is no API change on Resource Claim. However, a special resource claim is
+created to keep track of device allocations for extended resource. The special
+resource claim has following properties:
+
+  * It is namespace scoped, like other resource claims. Unlike resource slice,
```

Comment:
Just to clarify here, we are not talking about creating a new type right? This is just a special `ResourceClaim` in the sense that it is managed by the resource claim controller and lacks certain fields that would otherwise be expected on a ResourceClaim. Might be worth clarifying since this sentence about the scope of the resource can be a bit confusing.

Reply:
clarified in the KEP